.github/workflows/build-image.yml

```
@@ -162,16 +180,23 @@ jobs:
           platforms: ${{ matrix.platform }}
           push: true
           provenance: false
-          cache-from: type=gha,scope=cuda-${{ matrix.arch }}
-          cache-to: type=gha,mode=max,scope=cuda-${{ matrix.arch }}
-          labels: ${{ steps.meta.outputs.labels }}
-          outputs: >-
-            type=image,name=${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev,
-            push-by-digest=true,name-canonical=true,push=true
           build-args: |
+            TARGETOS=${{ matrix.os }}
+            BUILD_BASE_IMAGE_SUFFIX=${{ matrix.os == 'ubuntu' && 'ubuntu20.04' || matrix.base_image_suffix }}
+            FINAL_BASE_IMAGE_SUFFIX=${{ matrix.base_image_suffix }}
             VLLM_REPO=${{ inputs.vllm_repo || 'https://github.com/vllm-project/vllm.git' }}
             VLLM_COMMIT_SHA=${{ inputs.vllm_commit_sha || 'b8b302cde434df8c9289a2b465406b47ebab1c2d' }}
             VLLM_PRECOMPILED_WHEEL_COMMIT=${{ inputs.vllm_precompiled_wheel_commit || inputs.vllm_commit_sha || '' }}
+          cache-from: >-
+            type=gha,scope=cuda-dev-${{ matrix.os }}-${{
+            matrix.platform == 'linux/arm64' && 'arm64' || 'amd64' }}
+          cache-to: >-
+            type=gha,mode=max,scope=cuda-dev-${{ matrix.os }}-${{
+            matrix.platform == 'linux/arm64' && 'arm64' || 'amd64' }}
+          labels: ${{ steps.meta.outputs.labels }}
+          outputs: >-
+            type=image,name=${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev,
```

This needs to be scoped with the image suffix as well:

```suggestion
            type=image,name=${{ env.REGISTRY }}/${{ github.repository }}-cuda-dev${{matrix.base_image_suffix}},
```